protoconvert.go

```diff
@@ -123,20 +184,107 @@ func protoToValue(value *pb.Value) (any, error) {
 		if err != nil {
 			return nil, fmt.Errorf("failed to parse UUID: %v", err)
 		}
-		return &id, nil
+		return id, nil
 	case *pb.Value_Date:
 		d := time.Unix(int64((time.Duration(v.Date)*24*time.Hour)/time.Second), 0).UTC()
 		return &d, nil
 	case *pb.Value_Time:
-		t := time.Duration(v.Time)
-		return &t, nil
+		return time.Duration(v.Time), nil
 		// TODO: add decimal support
 		// TODO: add UDT support
 		// TODO: add varint support
-		// TODO: add map support
-		// TODO: add list support
-		// TODO: add set support
-		// TODO: add tuple support
 	}
 	return nil, fmt.Errorf("unsupported value type: %T, value: %+v", value.GetInner(), value.GetInner())
 }
+
+func protosToMap(values []*pb.Value, spec *pb.TypeSpec_Map) (any, error) {
+	l := len(values)
+	kts := spec.GetKey()
+	vts := spec.GetValue()
+
+	k0, err := protoToValue(values[0], kts)
```

Comment:
I think this might not handle empty maps correctly. It would panic?

Reply:
Fixed.